tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/RagConfig.java

```diff
@@ -0,0 +1,126 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.mailet.rag;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class RagConfig {
+    public static String AUTHORIZATION_TOKEN_PARAMETER_NAME = "ragondin.Token";
+    public static String TRUST_ALL_CERTIFICATES_PARAMETER_NAME = "ragondin.ssl.trust.all.certs";
+    public static String BASE_URL_PARAMETER_NAME = "ragondin.url";
+    public static String PARTITION_PATTERN_PARAMETER_NAME = "ragondin.partition.pattern";
+    public static final String DEFAULT_PARTITION_PATTERN = "{localPart}.twake.{domainName}";
+    private final String authorizationToken;
+    private final Optional<URL> baseURLOpt;
+    private final Boolean trustAllCertificates;
+    private final String partitionPattern;
+
+    public RagConfig(String authorizationToken, Boolean trustAllCertificates, Optional<URL> baseURLOpt, String partitionPattern) {
```

Comment:
```suggestion
    public RagConfig(String authorizationToken, boolean trustAllCertificates, Optional<URL> baseURLOpt, String partitionPattern) {
```

As it is not null let's use the primitive type.